lambdas/journeyengine/src/main/java/uk/gov/di/ipv/core/journeyengine/JourneyEngineHandler.java

```diff
@@ -142,34 +146,57 @@ private JourneyEngineResult executeJourneyEvent(
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    }
                     break;
                 case CRI_ADDRESS:
-                    updateUserState(CRI_FRAUD, ipvSessionItem);
-                    builder.setJourneyResponse(new JourneyResponse(criStartUri + FRAUD_CRI_ID));
+                    if (journeyStep.equals(NEXT_STEP)) {
+                        updateUserState(CRI_FRAUD, ipvSessionItem);
+                        builder.setJourneyResponse(new JourneyResponse(criStartUri + FRAUD_CRI_ID));
+                    } else {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    }
                     break;
                 case CRI_FRAUD:
-                    updateUserState(PRE_KBV_TRANSITION_PAGE, ipvSessionItem);
-                    builder.setPageResponse(new PageResponse(PRE_KBV_TRANSITION_PAGE.value));
+                    if (journeyStep.equals(NEXT_STEP)) {
+                        updateUserState(PRE_KBV_TRANSITION_PAGE, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PRE_KBV_TRANSITION_PAGE.value));
+                    } else {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    }
                     break;
                 case PRE_KBV_TRANSITION_PAGE:
                     updateUserState(CRI_KBV, ipvSessionItem);
                     builder.setJourneyResponse(new JourneyResponse(criStartUri + KBV_CRI_ID));
                     break;
                 case CRI_KBV:
-                    updateUserState(IPV_SUCCESS_PAGE, ipvSessionItem);
-                    builder.setPageResponse(new PageResponse(IPV_SUCCESS_PAGE.value));
+                    if (journeyStep.equals(NEXT_STEP)) {
+                        updateUserState(IPV_SUCCESS_PAGE, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(IPV_SUCCESS_PAGE.value));
+                    } else {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    }
                     break;
                 case IPV_SUCCESS_PAGE:
+                case CRI_ERROR:
                     builder.setJourneyResponse(new JourneyResponse(journeyEndUri));
                     break;
                 case DEBUG_PAGE:
                     builder.setPageResponse(new PageResponse(DEBUG_PAGE.value));
                     break;
-                case CRI_ERROR:
-                    builder.setPageResponse(new PageResponse(IPV_ERROR_PAGE.value));
-                    break;
+                default:
+                    updateUserState(CRI_ERROR, ipvSessionItem);
```

Comment:
Might be good to have some logging in here to show we've dropped out to the `default` case, which we probably should never actually do.